source/extensions/path/uri_template_lib/uri_template_internal.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "absl/container/flat_hash_set.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+#include "absl/types/variant.h"
+#include "re2/re2.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UriTemplate {
+
+namespace Internal {
+
+/**
+ * String to be concatenated in rewritten path.
+ */
+using Literal = absl::string_view;
+
+/**
+ * Determines what operations to use on the input pattern segment.
+ */
+enum class Operator { PathGlob, TextGlob };
+
+/**
+ * Represents a pattern variable. Variables are included in both path match and rewrite paths.
+ */
+struct Variable {
+  absl::string_view name_;
+
+  // replacement value for the rewrite
+  std::vector<absl::variant<Operator, Literal>> match_;
+
+  Variable(absl::string_view name, std::vector<absl::variant<Operator, Literal>> match)
```

Comment:
nit: Constructors and methods should come before members, per the style guide.

https://google.github.io/styleguide/cppguide.html#Declaration_Order

Reply:
changed